hipamd/src/hip_mrfs.cpp

```diff
@@ -0,0 +1,122 @@
+#include "hip_mrfs.hpp"
+#include <atomic>
+
+namespace hip {
+
+std::thread* hipMemCpyThread = nullptr;
```

Comment:
We need more than one background thread to do the work, otherwise there will be no concurrency.

We can create a vector of threads or #devices of threads for that.

Reply:
In order for hipStreamSync to work properly, we may need to have a stream<->thread mapping as well.